pkg/lockfile/parse-pipenv-lock.go

```diff
@@ -17,19 +17,19 @@ type PipenvLock struct {
 
 const PipenvEcosystem = PipEcosystem
 
-func ParsePipenvLock(pathToLockfile string) ([]PackageDetails, error) {
-	var parsedLockfile *PipenvLock
+type PipenvLockExtractor struct{}
 
-	lockfileContents, err := os.ReadFile(pathToLockfile)
+func (e PipenvLockExtractor) ShouldExtract(path string) bool {
+	return filepath.Base(path) == "Pipfile.lock"
```

Comment:
Practically no because I don't see a strong value in doing that now since it's not like it'll make the code faster or anything, and architecturally I'd argue you're conflating "extractor names" with "file paths"